Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -309,31 +321,49 @@ public CsharpCompiler(ErrorReporter reporter)
       Contract.Requires(dt != null);
       Contract.Requires(wr != null);
 
-      // public abstract class Dt<T> {  // for record types: drop "abstract"
+      // public interface _IDt<T> { // T has variance modifier
+      //   _IDt<T> _Get();  // for co-datatypes
+      //
+      //   bool is_Ctor0 { get; }
+      //   ...
+      //
+      //   T0 dtor_Dtor0 { get; }
+      //   ...
+      //
+      //   // Members that don't violate C# variance restrictions
+      // }
+      //
+      // public abstract class Dt<T> : _IDt<T> {  // for record types: drop "abstract"
       //   public Dt() { }
       //   #if TypeArgs.Count == 0
```

Comment:
Is this test really "#if there are no auto-init types in TypeArgs"?

Reply:
No, but it probably should. I'll address this in #1613.